tests/device_tests/test_msg_cardano_sign_transaction.py

```diff
@@ -17,215 +17,274 @@
+    ),
+    # transaction with stake deregistration
+    (
+        # network id
+        NETWORK_IDS["mainnet"],
+        # inputs
+        [SAMPLE_INPUTS["simple_input"]],
+        # outputs
+        [SAMPLE_OUTPUTS["simple_output"]],
+        # fee
+        42,
+        # ttl
+        10,
+        # certificates
+        [SAMPLE_CERTIFICATES["stake_deregistration"]],
+        # input flow
+        [[SWIPE, YES], [YES], [SWIPE, YES]],
+        # tx hash
+        "bda54c36b06db3f52853b6f700a87a4a3de1b5697e1bf4038538fa03cc3cd442",
+        # tx body
+        "83a500818258203b40265111d8bb3c3c608d95b3a0bf83461ace32d79336579a1939b3aad1c0b700018182582161a6274badf4c9ca583df893a73139625ff4dc73aaa3082e67d6d5d08e0ce3daa40102182a030a0481820182005820fcb95695ce37daf2daa2f961d02e05fee8a02be02b499608ea451aed76084e47a1008282582073fea80d424276ad0978d4fe5310e8bc2d485f5f6bb3bf87612989f112ad5a7d58408b2430cb72d59f7a64f29d3465845ede8377910e88f209fd004c91371018dbb2863babc7a9b62cca09525eb10effce0102dfb35655a85f29fd5b1c92e6af7f038258202c041c9c6a676ac54d25e2fdce44c56581e316ae43adc4c7bf17f23214d8d8925840f4623d7f81ef01304cd12289557158ceaae9c1e70e8f3d850f6a3fed715c3705c4cdd74ff49cb99a7d1c84845e0b842b802919589b6763d9219ba433776c3a00a0",
     ),
 ]
 
+# todo: GK - add invalid tests
+
 
 @pytest.mark.altcoin
 @pytest.mark.cardano
 @pytest.mark.skip_t1  # T1 support is not planned
 @pytest.mark.parametrize(
-    "protocol_magic,inputs,outputs,transactions,tx_hash,tx_body", VALID_VECTORS
+    "protocol_magic,inputs,outputs,fee,ttl,certificates,input_flow_sequence,tx_hash,tx_body",
+    VALID_VECTORS,
 )
+@pytest.mark.setup_client(mnemonic=SHELLEY_TEST_VECTORS_MNEMONIC)
 def test_cardano_sign_tx(
-    client, protocol_magic, inputs, outputs, transactions, tx_hash, tx_body
+    client,
+    protocol_magic,
+    inputs,
+    outputs,
+    fee,
+    ttl,
+    certificates,
+    input_flow_sequence,
+    tx_hash,
+    tx_body,
 ):
     inputs = [cardano.create_input(i) for i in inputs]
     outputs = [cardano.create_output(o) for o in outputs]
+    certificates = [cardano.create_certificate(c) for c in certificates]
 
-    expected_responses = [
-        messages.CardanoTxRequest(tx_index=i) for i in range(len(transactions))
-    ]
-    expected_responses += [
-        messages.ButtonRequest(code=messages.ButtonRequestType.Other),
-        messages.ButtonRequest(code=messages.ButtonRequestType.Other),
-        messages.CardanoSignedTx(),
-    ]
+    expected_responses = []
+    for i in range(len(input_flow_sequence)):
+        expected_responses.append(
+            messages.ButtonRequest(code=messages.ButtonRequestType.Other)
+        )
+
+    expected_responses.append(messages.CardanoSignedTx())
 
     def input_flow():
-        yield
-        client.debug.swipe_up()
-        client.debug.press_yes()
-        yield
-        client.debug.swipe_up()
-        client.debug.press_yes()
+        for sub_sequence in input_flow_sequence:
```

Comment:
I don't understand this. A sequence is composed of members / elements / terms, not subsequences. The names are not self-explanatory.